src/raven.js

```diff
@@ -469,6 +469,30 @@ Raven.prototype = {
         return this;
     },
 
+    /*
+     * Adds a shouldSendCallback callback that chains after any previously set
+     * callback(s).
+     *
+     * @param {function} callback The callback to run which allows
+     *                            introspecting the blob before sending
+     * @return {Raven}
+     */
+    addShouldSendCallback: function(callback) {
+        var prevCallback = this._globalOptions.shouldSendCallback;
+        if (prevCallback) {
+            this._globalOptions.shouldSendCallback = function() {
+                if (prevCallback && !prevCallback.apply(null, arguments)) {
```

Comment:
Also, I think it may be more prudent to invoke the callbacks with the current value of `this` (taken at the top of `addShouldSendCallback`) rather than pass `null`.